kernel/src/hil/touch.rs

```diff
@@ -0,0 +1,83 @@
+//! Interface for touch input devices
+
+use crate::ReturnCode;
+
+#[derive(Debug, Copy, Clone)]
+pub enum TouchStatus {
+    Pressed,
+    Released,
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum GestureEvent {
+    MoveUp,
+    MoveDown,
+    MoveLeft,
+    MoveRight,
+    ZoomIn,
+    ZoomOut,
+}
+
+#[derive(Copy, Clone)]
+pub struct TouchEvent {
+    pub status: TouchStatus,
+    // touch (x, y) position
+    pub x: usize,
+    pub y: usize,
+
+    // touch id, value defined by the driver
+    pub id: usize,
+
+    // touch area
+    pub area: Option<usize>,
+
+    // touch weight
+    pub weight: Option<usize>,
+}
+
+pub trait Touch {
+    // enable touch
+    fn enable(&self) -> ReturnCode;
+
+    // disable touch
+    fn disable(&self) -> ReturnCode;
+
+    fn set_client(&self, touch_client: &'static dyn TouchClient);
+}
+
+pub trait MultiTouch {
+    // enable touches
+    fn enable(&self) -> ReturnCode;
+
+    // disable touches
+    fn disable(&self) -> ReturnCode;
+
+    /// Returns the number of concurently supported touches
+    /// This function must be called in the same interrupt
+    /// as the event, otherwise data might not be available.
+    fn get_num_touches(&self) -> usize;
```

Comment:
This time-dependent bit makes me a bit nervous. Is it expensive to capture the number of touches, or is this something that could just be always included as part of the event?

Reply:
Done